app/reducers/threads.ts

```diff
@@ -0,0 +1,56 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { Thread } from 'app/actions/ActionTypes';
+import createEntityReducer from 'app/utils/createEntityReducer';
+import { mutateComments, selectCommentEntities } from './comments';
+import type { ID } from 'app/models';
+
+const mutate = mutateComments('threads');
+
+export default createEntityReducer({
+  key: 'threads',
+  types: {
+    fetch: Thread.FETCH_ALL,
+    mutate: Thread.CREATE,
+    delete: Thread.DELETE,
+  },
+  mutate,
+});
+
+export const selectThreadsByForumId = (forumId: ID) =>
+  createSelector(
+    (state) => state.threads.byId,
+    (state) => state.threads.items,
+    (threadsById, threadIds) => {
+      const filteredThreadIds = threadIds.filter((id) => {
+        const thread = threadsById[id];
+        return thread && thread.forum === forumId;
+      });
+
+      return filteredThreadIds.map((id) => threadsById[id]);
+    }
+  );
+
+export const selectThreads = createSelector(
+  (state) => state.threads.byId,
+  (state) => state.threads.items,
+  (threadsById, threadsIds) => threadsById.map((id) => threadsIds[id])
+);
```

Comment:
Seems like a funky way of doing it. I'd imagine it simpler to just add selectors that fetch threads from a given forum, instead of having to find the list of threads from the forum and then call the selector?